frontend/frontend-static-pages.php

```diff
@@ -278,4 +264,47 @@ protected function get_page_id( $query ) {
 
 		return 0; // No page queried.
 	}
+
+	/**
+	 * Add support for the theme customizer.
+	 *
+	 * @since 3.4.2
+	 *
+	 * @return void
+	 */
+	public function filter_customizer() {
+		add_filter( 'pre_option_page_on_front', array( $this, 'customize_page' ), 20 ); // After the customizer.
+		add_filter( 'pre_option_page_for_post', array( $this, 'customize_page' ), 20 );
+
+		add_filter( 'pll_pre_translation_url', array( $this, 'customize_translation_url' ), 20, 2 ); // After the generic hook in this class.
+	}
+
+	/**
+	 * Translate the page id when customized.
+	 *
+	 * @since 3.4.2
+	 *
+	 * @param int|false $pre A page id if the setting is customized, false otherwise.
+	 * @return int|false
+	 */
+	public function customize_page( $pre ) {
+		return is_numeric( $pre ) ? pll_get_post( (int) $pre ) : $pre;
+	}
+
+	/**
+	 * Fix the translation url the option 'show_on_front' is customized.
+	 *
+	 * @since 3.4.2
+	 *
+	 * @param string       $url      An empty string or the url of the translation of the current page.
+	 * @param PLL_Language $language The language of the translation.
+	 * @return string
+	 */
+	public function customize_translation_url( $url, $language ) {
+		if ( 'posts' === get_option( 'show_on_front' ) && is_front_page() ) {
+			// When the page on front displays posts, the home url is the same as the search url.
+			$url = $language->get_search_url();
```

Comment:
There is no need for this assignation, we can simply return the value.
```suggestion
			return $language->get_search_url();
```